mget_test.go

```diff
@@ -0,0 +1,304 @@
+			m := NewMGetRequest().AddDocs(tt.docs...)
+
+			require.Len(t, m.Docs, len(tt.want), "unexpected number of documents added to the request")
+
+			for i, gotDoc := range m.Docs {
+				wantDoc := tt.want[i]
+
+				require.Equal(t, wantDoc.id, gotDoc.ID())
+				require.Equal(t, wantDoc.index, gotDoc.Index())
+			}
+		})
+	}
+}
+
+func TestMGetRequest_MarshalJSON(t *testing.T) {
+	type fields struct {
+		Index string
+		Docs  []RoutableDoc
+	}
+	tests := []struct {
+		name    string
+		fields  fields
+		want    []byte
+		wantErr bool
+	}{
+		{
+			name: "Empty Request",
+			fields: fields{
+				Index: "",
+				Docs:  []RoutableDoc{},
+			},
+			want:    []byte(`{"docs":[]}`),
+			wantErr: false,
+		},
+		{
+			name: "Single document single type",
+			fields: fields{
+				Docs: []RoutableDoc{
+					testDoc{id: testID1, index: testIndex1},
+				},
+			},
+			want:    []byte(`{"docs":[{"_id":"test_id","_index":"test_index"}]}`),
+			wantErr: false,
+		},
+		{
+			name: "Multiple documents single type",
+			fields: fields{
+				Docs: []RoutableDoc{
+					testDoc{id: testID1, index: testIndex1},
+					testDoc{id: testID2, index: testIndex2},
+				},
+			},
+			want:    []byte(`{"docs":[{"_id":"test_id","_index":"test_index"},{"_id":"test_id2","_index":"test_index2"}]}`),
+			wantErr: false,
+		},
+		{
+			name: "Multiple documents mixed type",
+			fields: fields{
+				Docs: []RoutableDoc{
+					testDoc{id: testID1, index: testIndex1},
+					NewDocumentRef(testIndex2, testID2),
+				},
+			},
+			want:    []byte(`{"docs":[{"_id":"test_id","_index":"test_index"},{"_id":"test_id2","_index":"test_index2"}]}`),
+			wantErr: false,
+		},
+		{
+			name: "Document without index",
+			fields: fields{
+				Docs: []RoutableDoc{
+					testDoc{id: testID1},
+				},
+			},
+			want:    []byte(`{"docs":[{"_id":"test_id"}]}`),
+			wantErr: false,
+		},
+		{
+			name: "Document without id",
+			fields: fields{
+				Docs: []RoutableDoc{
+					testDoc{index: testIndex1},
+				},
+			},
+			want:    []byte(`{"docs":[{"_id":"","_index":"test_index"}]}`),
+			wantErr: false,
+		},
+		{
+			name: "Document without id and index",
+			fields: fields{
+				Docs: []RoutableDoc{
+					testDoc{},
+				},
+			},
+			want:    []byte(`{"docs":[{"_id":""}]}`),
+			wantErr: false,
+		},
+		{
+			name: "Request level index does not affect request json body",
+			fields: fields{
+				Index: testIndex2,
+				Docs: []RoutableDoc{
+					testDoc{id: testID1, index: testIndex1},
+				},
+			},
+			want:    []byte(`{"docs":[{"_id":"test_id","_index":"test_index"}]}`),
+			wantErr: false,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			m := &MGetRequest{
+				Index: tt.fields.Index,
+				Docs:  tt.fields.Docs,
+			}
+			got, err := m.MarshalJSON()
+			if (err != nil) != tt.wantErr {
+				t.Errorf("MarshalJSON() error = %v, wantErr %v", err, tt.wantErr)
+				return
+			}
+
+			require.Equalf(t, tt.want, got, "expected %s - got %s", tt.want, got)
```

Comment:
Should use [JSONEqf](https://pkg.go.dev/github.com/stretchr/testify@v1.8.1/require#JSONEqf) to assert on JSON structures